study-group/Toast-Implementation-Studies.md

```diff
@@ -70,25 +70,190 @@ The `<paper-toast>` holds onto information about how it was dismissed the last t
 
-## Material-UI Snackbar
+## [Material-UI Snackbar](https://material-ui.com/api/snackbar/)
 
 *TODO: Fill in study for this library*
 
-## Android Toast
+## [Android `Toast`](https://developer.android.com/guide/topics/ui/notifiers/toasts)
+Android `Toast` is a simple toast implementation built by the Android team for usage in Android apps.
 
-*TODO: Fill in study for this library*
+### Design & Sample Code
+**Note**: Since Android development is done in Kotlin all code samples will be Kotlin.
 
-## react-toastify
+To trigger displaying a simple Android `Toast`, put this in your Kotlin:
 
-*TODO: Fill in study for this library*
+```kotlin
+val text = "Hello toast!"
+val duration = Toast.LENGTH_SHORT
+
+val toast = Toast.makeText(applicationContext, text, duration)
+toast.show()
+```
+
+This creates a toast object filled with a context, some text, and a duration, which is triggered to display by the `toast.show()` method.
+This will show up on the bottom of the screen by default.
+To control the position of the toast, set the gravity, like this:
+
+```kotlin
+toast.setGravity(Gravity.TOP or Gravity.LEFT, 0, 0)
+```
+
+This sample will move the toast to the top left of the screen.
+The final two parameters are used to offset the toast from its center of gravity, so this:
+
+```kotlin
+toast.setGravity(Gravity.TOP or Gravity.LEFT, 10, 10)
+```
+
+Will show the toast further down and further right.
+The toast can also be injected with a custom layout, which requires more complex Kotlin, explained [here](https://developer.android.com/guide/topics/ui/notifiers/toasts#CustomToastView).
+The `text`, `view`, `gravity`, and `duration` of the toast are all the elements the developer can control when working with this very straightforward component.
+
+
+### Notable Features & Details
+
+#### Gravity
+The library takes the Android approach to positioning,
+by letting the user define the Gravity of the element,
+along with offset values for the axes.
+
+#### Duration Constants
+The library comes with baked in constants for displaying the toast for long and short periods of time.
+
+### Takeaways
+- High opinion, low control.
+- The API for the `Toast` object provides getters and setters for each of `text`, `view`, `gravity`, `offset` and `duration`.
+- Beyond getters and setters, the only control the object allows is `.makeText()`, `.show()` and `.cancel()`.
+
+## [react-toastify](https://github.com/fkhadra/react-toastify)
+`react-toastify` is a React component library by author [@fkhadra](https://github.com/fkhadra) for rich, extensible notification toasts.
+
+### Design & Sample Code
+
+To use a `react-toastify` toast requires some simple JavaScript:
+
+```js
+class App extends Component {
+  notify = () => toast("Very simple toast!");
+
+  render(){
+    return (
+      <div>
+        <button onClick={this.notify}>Notify !</button>
+        <ToastContainer />
```

Comment:
Maybe show the importing of toast/ToastContainer?

Reply:
Good call